### ISSUE TITLE
v2.x plans and goals

### ISSUE THREAD

**Issue by vlkosinov** [dailymotion/vast-client-js#185]
Hi there! I noticed that you have some work in v2.x branch. Do you have tasks list for new release or something like? 

We are using this module in our apps at @SPBTV. So we would be happy to help with the development of new features.

**Comment by michaelvial** [comment]
Hi @vlkosinov,

Sorry for the late reply. Yeah, we've created a `v2` branch, we're actually using it on dailymotion.com. At the moment, it's quite similar to the `master` one, I think the only breaking change is that the parser always returns an array even when VAST file is empty.

Last year, our ad integration was relying on one sole VPAID manager. The VAST Client was barely used at all. Things changed in January, as we moved to a VMAP/VAST integration. The VAST Client is now a key component in our current stack.
This has helped us to fix a lot of issues in this library. But, as we process way more VAST files than we ever did, we have encountered some problems that cannot be resolved without causing a breaking change. So, we're going to fix these in `v2.x`, and once we 're good, we'll make a proper `2.0.0` version.

The main issues that cause us trouble are:
- Loss of wrapper informations. For example, if both `<wrapper>` and `<inline>` provide a creative `adId`, the `ad` object returned by the parser will only have the `<inline>` `adId`.
- All `VASTAdTagURI` are resolved when the parser is called. Imagine having a VAST file containing two `<ad>` tag, each one being a `<wrapper>`. If the 1st one fills, we shouldn't call the 2nd.
- No support of ad fallback. Following our previous example, if we can't play the 1st ad, we need a method to resolve/retrieve the 2nd ad. 

Ideally we would also like to:
- decaffeinate and move to `es6`
- add a type checker?
